app/jobs/regular/change_display_name.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ChangeDisplayName < ::Jobs::Base
+    sidekiq_options queue: "low"
+
+    def execute(args)
+      @user = User.find_by(id: args[:user_id])
+
+      return unless user.present?
+
+      # We need to account for the case where the instance allows
+      # name to be empty by falling back to username.
+      @old_display_name = (args[:old_name].presence || user.username).unicode_normalize
+      @new_display_name = (args[:new_name].presence || user.username).unicode_normalize
+
+      @quote_rewriter = QuoteRewriter.new(user.id)
+
+      update_posts
+      update_revisions
+    end
+
+    private
+
+    attr_reader :user, :old_display_name, :new_display_name, :quote_rewriter
+
+    def update_posts
+      Post
+        .with_deleted
+        .joins(quoted("posts.id"))
```

Comment:
In `quoted`, we join against `quoted_posts` and then against `posts`. It seems like there is no need to join against the `posts` table when we're already querying against the posts table

Reply:
I assume this is because the same method is used when querying `post_revisions`. So perhaps this method was extracted a bit prematurely. I will inline the joins in the respective methods instead.

Reply:
Ah. I figured it out now. There are two sides to it. One side is the side that's doing the quoting, and the other side is the post that gets quoted, and quoted_posts acts as the join table.

We need the quoting post where the quoted post user is a certain ID, or in other words we need both sides of posts, hence the extra join.